### PR TITLE
KAFKA-4361: Streams does not respect user configs for "default" params

### PR DESCRIPTION
**Pull request by dguy**
Enable user provided consumer and producer configs to override the streams default configs.

### CODE REVIEW

**Comment by dguy** [comment]
@theduderog @guozhangwang @enothereska @mjsax

**Comment by enothereska** [comment]
LGTM

**Comment by mjsax** [comment]
LGTM. Thanks for adding tests!

**Comment by theduderog** [comment]
testing now.  thanks, @dguy

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -362,16 +365,18 @@ public StreamsConfig(Map<?, ?> props) {
...
 
         // disable auto commit and throw exception if there is user overridden values,
         // this is necessary for streams commit semantics
-        if (consumerProps.containsKey(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG)) {
+        if (clientProvidedProps.containsKey(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG)) {
             throw new ConfigException("Unexpected user-specified consumer config " + ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG
                     + ", as the streams client will always turn off auto committing.");
         }
 
-        consumerProps.putAll(CONSUMER_DEFAULT_OVERRIDES);
```
Comment (theduderog) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:374]: The restore consumer needs to override one param:

`consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");`

**Comment by theduderog** [comment]
I think the restoration consumer should start from "earliest" even if the application defaults to "latest".  Would be great to have a test case for this.

**Comment by theduderog** [comment]
It may be that the restoration consumer ignore this setting.

**Comment by dguy** [comment]
@theduderog the restore consumer doesn't ignores this setting.

**Comment by theduderog** [comment]
if the restoration consumer doesn't ignore the setting, then I think it should be overridden for the restoration consumer.

**Comment by dguy** [comment]
sorry - that was my bad typing this morning. The restore consumer ignores the setting

**Comment by theduderog** [comment]
awesome.  thanks

**Comment by mjsax** [comment]
@dguy You should ☕️  yourself before any ⌨️  :P

**Pull request closed without merging**